website/for_teachers.py

```diff
@@ -1075,10 +1075,10 @@ def update_adventure(self, user):
         elif current_adventure["creator"] != user["username"]:
             return gettext("unauthorized"), 403
         current_classes = {}
-        if body.get("classes"):
+        if body.get("classes") and current_adventure.get('classes'):
             current_classes = current_adventure["classes"]
```

Comment:
Imagine `current_adventure` being an existing adventure. It most likely doesn't have classes, therefore the next line should be changed. 

```suggestion
        if body.get("classes"):
            current_classes = current_adventure.get('classes', [])
```

Same goes for the levels you made below and all other places you use classes as well!

Reply:
These are the only places